crates/aptos-compression/src/metrics.rs

```diff
@@ -62,35 +39,89 @@ pub static OPERATION_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
     )
     .unwrap()
 });
 
 /// Increments the compression byte count based on the given data type
-pub fn increment_compression_byte_count(
+fn increment_compression_byte_count(
+    operation: &str,
     data_type: &str,
-    client: CompressionClient,
+    client: &CompressionClient,
     byte_count: u64,
 ) {
     BYTE_COUNTS
-        .with_label_values(&[data_type, client.get_label()])
+        .with_label_values(&[operation, data_type, client.get_label()])
         .inc_by(byte_count)
 }
 
 /// Increments the compression error count based on the given operation
-pub fn increment_compression_error(operation: &str, client: CompressionClient) {
+pub fn increment_compression_error(client: &CompressionClient) {
+    increment_error_count(COMPRESS, client)
+}
+
+/// Increments the decompression error count based on the given operation
+pub fn increment_decompression_error(client: &CompressionClient) {
+    increment_error_count(DECOMPRESS, client)
+}
+
+/// Increments the error count based on the given operation
+fn increment_error_count(operation: &str, client: &CompressionClient) {
     ERROR_COUNTS
         .with_label_values(&[operation, client.get_label()])
         .inc()
 }
 
-/// Starts the timer for the compression operation using the label
-pub fn start_compression_operation_timer(
-    operation: &str,
-    client: CompressionClient,
-) -> HistogramTimer {
+/// Starts the timer for the compression operation
+pub fn start_compression_operation_timer(client: &CompressionClient) -> HistogramTimer {
+    start_operation_timer(COMPRESS, client)
+}
+
+/// Starts the timer for the decompression operation
+pub fn start_decompression_operation_timer(client: &CompressionClient) -> HistogramTimer {
+    start_operation_timer(DECOMPRESS, client)
+}
+
+/// Starts the timer for the given operation
+fn start_operation_timer(operation: &str, client: &CompressionClient) -> HistogramTimer {
     OPERATION_LATENCY
         .with_label_values(&[operation, client.get_label()])
         .start_timer()
 }
+
+/// Updates the compression metrics for the given data sets
+pub fn update_compression_metrics(
+    client: &CompressionClient,
+    raw_data: &Vec<u8>,
+    compressed_data: &Vec<u8>,
+) {
+    update_operation_metrics(COMPRESS, client, raw_data, compressed_data);
+}
+
+/// Updates the decompression metrics for the given data sets
+pub fn update_decompression_metrics(
+    client: &CompressionClient,
+    compressed_data: &Vec<u8>,
+    raw_data: &Vec<u8>,
+) {
+    update_operation_metrics(DECOMPRESS, client, raw_data, compressed_data);
+}
+
+/// Updates the operation metrics based on the given data
+/// (e.g., raw and compressed data sizes).
+fn update_operation_metrics(
+    operation: &str,
+    client: &CompressionClient,
+    raw_data: &Vec<u8>,
+    compressed_data: &Vec<u8>,
+) {
+    increment_compression_byte_count(operation, RAW_BYTES, client, raw_data.len() as u64);
```

Comment:
Wonder if the compression ratio is an interesting metric as well? I see you've removed it (although it was only used for trace).

(No strong opinion here, it just crossed my mind)

Reply:
Aah, we're still tracking the ratio -- the logic I removed was for the `trace!` logs, which we never use. In grafana we're tracking the ratio by dividing the raw bytes by the compressed bytes, so nothing should be lost 😄

Reply:
Oh ok, I was actually thinking of possible outliers -- if some data compresses too well or not well at all.